### PR TITLE
Added scanTypeNames() using FastClasspathScanner Library

### PR DESCRIPTION
**Pull request by abibell**
JaversBuilder now supports scanTypeNames() using FastClasspathScanner Library. Default implementations of JaversMongoAutoConfiguration and JaversSqlAutoConfiguration support package scanning using `packagesToScan` configuration property. `packagesToScan` supports multiple values separated by comma

Resolving https://github.com/javers/javers/issues/263

### CODE REVIEW

File: javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java [hunk 1]
```diff
@@ -1,15 +1,14 @@
 package org.javers.spring.boot.sql;
 
-import com.google.common.collect.Maps;
```
Comment (bartoszwalacik) [inline javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java]: please don't commit unnecessary changes, it makes code review harder

File: javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java [hunk 2]
```diff
@@ -22,7 +21,7 @@ public CommitPropertiesProvider commitPropertiesProvider() {
         return new CommitPropertiesProvider() {
             @Override
             public Map<String, String> provide() {
-                Map props = new HashMap();
+                Map<String, String> props = new HashMap<String, String>();
```
Comment (bartoszwalacik) [inline javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java]: please don't commit changes unrelated to main issue

File: javers-core/src/main/java/org/javers/core/JaversBuilder.java [hunk 3]
```diff
@@ -233,22 +233,19 @@ public JaversBuilder registerValueObject(ValueObjectDefinition valueObjectDefini
...
      */
-    public JaversBuilder scanTypeNames(String packageToScan){
-        throw new RuntimeException("JaversBuilder.scanTypeNames(String packageToScan) is not implemented! " +
-                "If you think that this method should be implemented, " +
-                "vote here: https://github.com/javers/javers/issues/263");
+    public JaversBuilder scanTypeNames(String packagesToScan){
+    	if(packagesToScan==null || packagesToScan.trim().isEmpty())
+    		return this;
+		
+		Class<?>[] list = ReflectionUtil.getClasses(TypeName.class, packagesToScan.split(","));
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/JaversBuilder.java]: as you can see, indentation is messed up (use spaces, not tabs)

File: javers-core/src/main/java/org/javers/core/JaversBuilder.java [hunk 4]
```diff
@@ -233,22 +233,19 @@ public JaversBuilder registerValueObject(ValueObjectDefinition valueObjectDefini
...
-     *
-     * If you think that this method should be implemented,
-     * vote here: <a href="https://github.com/javers/javers/issues/263">issue/263</a>
      */
-    public JaversBuilder scanTypeNames(String packageToScan){
-        throw new RuntimeException("JaversBuilder.scanTypeNames(String packageToScan) is not implemented! " +
-                "If you think that this method should be implemented, " +
-                "vote here: https://github.com/javers/javers/issues/263");
+    public JaversBuilder scanTypeNames(String packagesToScan){
+    	if(packagesToScan==null || packagesToScan.trim().isEmpty())
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/JaversBuilder.java]:
```
if (...) {
    return ...
}
```

File: javers-core/src/main/java/org/javers/common/reflection/ReflectionUtil.java [hunk 5]
```diff
@@ -174,6 +184,21 @@ private static boolean isProtected(Member member){
...
+    	for (String c : names) {
+    		Class<?> classType;
+			try {
+				classType = Class.forName(c);
+			} catch (ClassNotFoundException e) {
+				throw new JaversException(JaversExceptionCode.CANT_FIND_CLASS_WITH_TYPENAME, c);
+			}
+    		classes.add(classType);
+    	}
+    	return null;
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/common/reflection/ReflectionUtil.java]: this method (clue of your PR) always returns null.
Write tests, check if they fail, and then implement a new feature.

File: javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java (hunk 1 above)
Comment (abibell) [inline javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java]: Yup removing some of warnings in code is unrelated

File: javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java (hunk 2 above)
Comment (abibell) [inline javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java]: Yup removing some of warnings in code is unrelated

**Comment by bartoszwalacik** [comment]
we will continue work on abibell-master branch (https://github.com/javers/javers/tree/abibell-master). I'm going to push some tests

**Comment by abibell** [comment]
I addressed some feedback. I realised that my Eclipse didn't recognize groovy files. Anyway its all yours. Revert commits, do whatever you want on it.

**Comment by bartoszwalacik** [comment]
merged from another branch https://github.com/javers/javers/pull/434
this feature will be released in JaVers 2.3 (tomorrow)
